### DISCUSSION TITLE
Speed sensor error

### DISCUSSION THREAD

**Discussion by Energeek136** [danielnilsson9/bbs-fw#229] [General]
Is it possible to implement that when magnet comes off the wheel that display shows error and doesnt give any power?

**Comment by dav0000000** [comment]
I'll start by asking - why? Anything is possible.

**Reply by Energeek136** [reply]
If you have a bike that is limited to 25km/h legal limit, and the speed sensor doesnt get any signal(shows 0 at screen, magnet misalignment) then the bike will have nonlimited speed greater then 25.

The tsdz2 has this, if the speed sensor doesnt get signal from magnet then the display shows 99km/h and motor wont give any power

**Reply by dav0000000** [reply]
Ok interesting. So it's for legal purposes. I didn't think of that but you are correct that law enforcement could possibly use that against you but I'd just show them that the speed sensor is working if they challenge you. This firmware does not read the crank speed which would be the best way to implement this feature along with a timer. Daniel said it is possible to read the crank speed. I would find it useful having this info but I don't have the skill to implement it.

**Reply by Paulii6661** [reply]
Oh well you can just change wheel size and go faster I don't think police would give a damn at least in my country.

Also remember you can use normal and "sport" mode with this firmware by enabling it to going to 0 pas turn in lights and then it's activated. So you can have 1-9 set up for legal limit but as long as you go off road or feel comfortable you can switch to illegal fun drive.. :)

**Comment by WvBijsterveld** [comment]
As far as I know in the original firmware this is applied, however I don’t know if in the bbsfw firmware it is added or just removed. Meaning you can search the comments inside the source files mentioning no magnet scenario that it still gives power or does NOT execute the error. Otherwise you could write the code for it yourself, Im a starting developer and it should be pretty easy, easy enough that I think chatgpt (or any other ai tool) will write it good for you on the first try. Just give enough context about everything and try to read the source files so you have a good understanding first.

**Reply by Energeek136** [reply]
Yes my bad the original firmware does show error but only briefly I think and  after 20min the whole display shuts down. But still you can drive it like that unlimited

**Reply by dav0000000** [reply]
You could start a timer when target_current is greater than 0 and throw the error if no speed is found after 8 seconds.

**Comment by WvBijsterveld** [comment]
that shouldn’t be anything to do with it. with less grip it can slip more easily but that won’t be that benificatial as acceleration should still be the same effort.
